src/state/file-browser/actions.js

```diff
@@ -34,6 +35,42 @@ const wrapApiCall = apiFunc => (...args) => async (dispatch) => {
 
 
 // thunks
+
+const getBase64 = file => (
+  new Promise((resolve) => {
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.onload = () => {
+      const base64 = reader.result.split(',');
+      resolve(base64[1]);
+    };
+  }));
+
+export const fetchFile = (filename, extensions = ['.txt']) => (dispatch, getState) =>
+  new Promise((resolve) => {
+    const state = getState();
+    const file = filename.split('.');
+    if (extensions.includes(`.${file[1]}`)) {
+      dispatch(toggleLoading('file'));
+      const { protectedFolder, currentPath } = getPathInfo(state);
+      const queryString = `?protectedFolder=${protectedFolder === null ? false : protectedFolder}&currentPath=${currentPath}/${filename}`;
+      getFile(queryString).then((response) => {
+        response.json().then((json) => {
+          if (response.ok) {
+            dispatch(initialize('CreateTextFileForm', { content: window.atob(json.payload.base64) }));
+          } else {
+            dispatch(addErrors(json.errors.map(e => e.message)));
+          }
+          dispatch(toggleLoading('file'));
+          resolve();
+        });
+      });
+    } else {
+      dispatch(addToast('Error file extension'), TOAST_ERROR);
```

Comment:
are we still displaying the form to edit the file at this point? if that is the case we should probably redirect or give the means to redirect.